### DISCUSSION TITLE
I want to know the connections and features of COMPASS, dolphin, s1-reader, sweets, and fringe.

### DISCUSSION THREAD

**Discussion by ZGHHGZ** [isce-framework/dolphin#97] [Q&A]
### Checked for duplicates

Yes - I've already checked

### Alternatives considered

Yes - and alternatives don't suffice

### Related problems

I want to know the connections and features of COMPASS, dolphin, s1-reader, sweets, and fringe.

### Describe the feature request

I need or want [...]

**Comment by scottstanie** [comment]
- `COMPASS` create geocoded SLCs for Sentinel-1 using `isce3`
- `s1-reader` is the tool that reads the Sentinel-1 metadata for each burst. It's used by both `COMPASS` and the [RTC](https://github.com/opera-adt/RTC) repo
- Fringe was the prototype software for doing SqueeSAR-like PS/DS processing. `dolphin` is a new version of this, trying to improve on general software stability/usability so it can be run in production at scale.
- `sweets` is an early experiment in making an easy-to-use workflows with these new tools. Like [topsStack](https://github.com/isce-framework/isce2/tree/main/contrib/stack/topsStack) but for isce3 (and with geocoded SLCs, not radar coordinates). It will probably have an option to do `dolphin` processing, but for now it just makes multilooked interferograms to be ready for MintPy

**Comment by shitong01** [comment]
following this, are there any plans to  incoporate tools in sweets to create a co-registered stack from NISAR RSLCs?

**Reply by scottstanie** [reply]
I think that could be brought up on the `sweets` repo, as `dolphin` definitely does not have plans to coregister any RSLCs.

However, you might get more of a response asking `isce3`, as the current plan is to have the NISAR production workflows serve as examples for how users can build their own custom workflows with the `isce3` core libary. I haven't tried to make a radar-coordinate stack processing, but you might use something like https://github.com/isce-framework/isce3/blob/develop/python/packages/nisar/workflows/resample_slc_v2.py in a loop over SLCs

**Reply by shitong01** [reply]
Thanks @scottstanie, this has been helpful! Yes, now looking at the ISCE3 workflow repos it seems like most of the functions for coregistration lies there.

It would be great though if the sweets repo has a wrapper for this since it encompasses downloads as well.
